### PR TITLE
feat: linting yaml files

### PR DESCRIPTION
**Pull request by tomMoulard**
This PR adds yaml linting, as well as linting all yaml files in the project.

See [yamllint](https://github.com/adrienverge/yamllint) for more information.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,6 +31,7 @@ mod: ## go mod tidy
 inst: ## go install tools
 	$(call print-target)
 	cd tools && go install $(shell cd tools && go list -e -f '{{ join .Imports " " }}' -tags=tools)
+	pip install --user yamllint
```
Comment (pellared) [inline Makefile:34]: Not everyone has python and wants to install yamllint globally.

How about installing it via docker in a virtual-env? 

Here is an example how we install and use a Python tool in OpenTelemetry Go: https://github.com/open-telemetry/opentelemetry-go/blob/f88533381ba23b17bb33f22a762053fb35fd80e7/Makefile#L241-L243

File: Makefile [hunk 2]
```diff
@@ -51,6 +52,7 @@ spell: ## misspell
 .PHONY: lint
 lint: ## golangci-lint
 	$(call print-target)
+	yamllint .
```
Comment (pellared) [inline Makefile:55]: I suggest splitting `lint` to two targets: `golint` and `yamllint`. We can still have `lint` which would depend on both of them.

**Comment by pellared** [comment]
After thinking a while I am not think it is worth to add this to this template (I want to keep it pretty minimalistic). You can still add an issue. 

Personally, I like https://github.com/igorshubovych/markdownlint-cli very much but I decided to not add it to this template.

**Comment by tomMoulard** [comment]
For sure, I get it.

Thanks anyway for your awesome project :100:

**Pull request closed without merging**